packages/npm/financekit/ios/Financekit.swift

```diff
@@ -307,31 +300,29 @@ final class Financekit: NSObject {
     
     @objc(accountBalances:withResolve:withRejecter:)
     func accountBalances(stringifiedQuery: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
+        guard #available(iOS 17.4, *) else {
+            return reject(CODE_LOW_OS_VERSION, MESSAGE_LOW_OS_VERSION, nil)
+        }
         Task {
             do {
                 guard let queryData: Data = stringifiedQuery.data(using: .utf8) else {
                     return reject(CODE_QUERY_INVALID, MESSAGE_UNKNOWN_PARAMS_ISSUE, nil)
                 }
                 let queryObject = try JSONDecoder().decode(FinanceKitQuery.self, from: queryData)
                 let query = AccountBalanceQuery(sortDescriptors: [], predicate: nil, limit: queryObject.limit, offset: queryObject.offset)
-                let balances = try await store.accountBalances(query: query)
+                let balances = try await FinanceStore.shared.accountBalances(query: query)
                 let financekitBalances = try balances.map { accountBalance in
                     let currentBalance = try FinanceKitCurrentBalance(currentBalance: accountBalance.currentBalance)
                     return FinanceKitAccountBalance(id: accountBalance.id, accountID: accountBalance.accountID, currentBalance: currentBalance)
                 }
                 
-                let data = try jsonEncoder.encode(balances)
+                let data = try jsonEncoder.encode(financekitBalances)
                 guard let stringifiedResponse = String(data: data, encoding: .utf8) else {
                     return reject(CODE_ACCOUNT_BALANCES_INVALID, MESSAGE_UNKNOWN_RESPONSE_ISSUE, nil)
                 }
                 resolve(stringifiedResponse)
-            } catch let error as FinancekitError {
-                switch error {
-                case .unknownCurrentBalanceType:
-                    reject(CODE_UNKNOWN_CURRENT_BALANCE_TYPE, error.message, error)
-                default:
-                    reject(CODE_UNKNOWN, error.message, error)
-                }
+            } catch let error as UnknownCurrentBalanceTypeError {
```

Comment:
I agree with what you're trying to accomplish with this change (eliminating impossible code paths) but Swift actually has syntax for this that would work with the original enum: 
```suggestion
            } catch FinancekitError.unknownCurrentBalanceType(let message). {
```

Reply:
I think mostly this change was due to not seeing a point in having an enum for all possible errors when functions can only return one or two of them. Especially when that enum isn't exposed to end users or used outside of swift.